src/WebJobs.Script/Diagnostics/OpenTelemetry/TraceFilterProcessor.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using OpenTelemetry;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics.OpenTelemetry
+{
+    internal class TraceFilterProcessor : BaseProcessor<Activity>
+    {
+        private TraceFilterProcessor() { }
+
+        public static TraceFilterProcessor Instance { get; } = new TraceFilterProcessor();
+
+        public override void OnEnd(Activity data)
+        {
+            var dataTags = data.Tags.ToImmutableDictionary();
```

Comment:
Can we avoid this call? It looks very expensive to be in a telemetry processor. Avoid allocations as much as possible. I recommend looking into OTel's code, they have some high-performance way to interacting with tags I believe.

Reply:
Removed ToImmutableDictionary.